hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/entity/impl/ReactiveAbstractEntityPersister.java

```diff
@@ -728,19 +726,20 @@ else if ( !delegate().isModifiableEntity( entry ) ) {
 		}
 
 		// Now update only the tables with dirty properties (and the table with the version number)
-		return loop(
-				IntStream.range(0, span).filter( i-> tableUpdateNeeded[i] ),
-				table -> updateOrInsertReactive(
-						id,
-						fields,
-						oldFields,
-						table == 0 ? rowId : null,
-						propsToUpdate,
-						table,
-						oldVersion,
-						updateStrings[table],
-						session
-				)
+		return loop( 0, span,
+				table -> tableUpdateNeeded[table]
+						? updateOrInsertReactive(
+								id,
+								fields,
+								oldFields,
+								table == 0 ? rowId : null,
+								propsToUpdate,
+								table,
+								oldVersion,
+								updateStrings[table],
+								session
+						)
+						: voidFuture()
```

Comment:
I'm not convinced by this one. I'm not sure what implication passing all those extra `voidFuture()`s has. Perhaps it's fine, but I would need to dig a lot deeper into the code.